CMakeLists.txt

```diff
@@ -283,43 +315,175 @@ if(UNIX AND NOT APPLE)
+  add_dependency_includes(${MASON_PACKAGE_stxxl_INCLUDE_DIRS})
+  set(STXXL_LIBRARY ${MASON_PACKAGE_stxxl_STATIC_LIBS})
+
+  mason_use(expat VERSION ${MASON_EXPAT_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_expat_INCLUDE_DIRS})
+  set(EXPAT_LIBRARIES ${MASON_PACKAGE_expat_STATIC_LIBS})
+
+  mason_use(lua VERSION ${MASON_LUA_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_lua_INCLUDE_DIRS})
+  set(USED_LUA_LIBRARIES ${MASON_PACKAGE_lua_STATIC_LIBS})
+
+  mason_use(luabind_lua524 VERSION ${MASON_LUABIND_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_luabind_lua524_INCLUDE_DIRS})
+  set(LUABIND_LIBRARY ${MASON_PACKAGE_luabind_lua524_STATIC_LIBS})
+
+  mason_use(bzip2 VERSION ${MASON_BZIP2_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_bzip2_INCLUDE_DIRS})
+  set(BZIP2_LIBRARIES ${MASON_PACKAGE_bzip2_STATIC_LIBS})
+
+  # todo - system packages do not provide a remote tarball
+  # but the mason-cmake integration expects one
+  #mason_use(zlib VERSION system)
+  #add_dependency_includes(${MASON_PACKAGE_zlib_INCLUDE_DIRS})
+  #set(ZLIB_LIBRARY ${MASON_PACKAGE_zlib_LDFLAGS})
+  set(ZLIB_LIBRARY "-lz")
+
+  mason_use(tbb VERSION ${MASON_TBB_VERSION})
+  add_dependency_includes(${MASON_PACKAGE_tbb_INCLUDE_DIRS})
+  set(TBB_LIBRARIES ${MASON_PACKAGE_tbb_LDFLAGS})
+
+  if(NOT MASON_PACKAGE_tbb_LIBRARY_DIRS)
+    message(FATAL_ERROR "MASON_PACKAGE_tbb_LIBRARY_DIRS is empty, rpath will not work")
+  endif()
+  foreach(libpath ${MASON_PACKAGE_tbb_LIBRARY_DIRS})
+    if(UNIX AND NOT APPLE)
+       set(LINKER_FLAGS "-Wl,-rpath,${libpath}")
+    elseif(APPLE)
+       set(LINKER_FLAGS "-Wl,-rpath, -Wl,${libpath}")
+    endif()
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LINKER_FLAGS}")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINKER_FLAGS}")
+    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${LINKER_FLAGS}")
+    file(GLOB TBBGlob ${libpath}/*.*)
+    install(FILES ${TBBGlob} DESTINATION lib)
+  endforeach()
+
+  include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/include)
+
+  if(BUILD_COMPONENTS)
+    message(FATAL_ERROR "BUILD_COMPONENTS is not supported with ENABLE_MASON")
+  endif()
+
+  # current mason packages target -D_GLIBCXX_USE_CXX11_ABI=0
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
+
+  if(ENABLE_CCACHE)
+    mason_use(ccache VERSION ${MASON_CCACHE_VERSION})
+    message(STATUS "Setting ccache to ccache ${MASON_CCACHE_VERSION} (via mason) ${MASON_PACKAGE_ccache_PREFIX}/bin/ccache")
+    message(STATUS "Using ccache to speed up incremental builds")
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${MASON_PACKAGE_ccache_PREFIX}/bin/ccache)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ${MASON_PACKAGE_ccache_PREFIX}/bin/ccache)
+    set(ENV{CCACHE_CPP2} "true")
+  endif()
+
+
+else()
+
+  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/cmake")
```

Comment:
Moved outside of the mason/non-mason blocks. Now below them: https://github.com/Project-OSRM/osrm-backend/commit/87d09f78e6584037b6e879705e6b6a65ecabd3a8#diff-af3b638bc2a3e6c650974192a53c7291R459